nisystemlink/clients/auth/models/_auth_models.py

```diff
@@ -0,0 +1,373 @@
+    The unique id
+    """
+    first_name: Optional[str] = Field(
+        None, alias="firstName", example="user-first-name"
+    )
+    """
+    The user's first name
+    """
+    last_name: Optional[str] = Field(None, alias="lastName", example="user-last-name")
+    """
+    The user's last name
+    """
+    email: Optional[str] = Field(None, example="example@email.com")
+    """
+    The user's email
+    """
+    phone: Optional[str] = Field(None, example="555-555-5555")
+    """
+    The user's contact phone number
+    """
+    niua_id: Optional[str] = Field(None, alias="niuaId", example="example@email.com")
+    """
+    The external id (niuaId, SID, login name)
+    """
+    login: Optional[str] = None
+    """
+    The login name of the user. This the "username" or equivalent entered when
+    the user authenticates with the identity provider.
+    """
+    accepted_to_s: Optional[bool] = Field(None, alias="acceptedToS", example=True)
+    """
+    (deprecated) Whether the user accepted the terms of service
+    """
+    properties: Optional[Dict[str, str]] = Field(None, example={"key1": "value1"})
+    """
+    A map of key value properties
+    """
+    keywords: Optional[List[str]] = None
+    """
+    A list of keywords associated with the user
+    """
+    created: Optional[datetime] = Field(None, example="2019-12-02T15:31:45.379Z")
+    """
+    The created timestamp
+    """
+    updated: Optional[datetime] = Field(None, example="2019-12-02T15:31:45.379Z")
+    """
+    The last updated timestamp
+    """
+    org_id: Optional[str] = Field(None, alias="orgId", example="org-id")
+    """
+    The id of the organization
+    """
+    policies: Optional[List[str]] = None
+    """
+    A list of policy ids to reference existing policies
+    """
+    status: Optional[Status] = Field(None, example="active")
+    """
+    The status of the users' registration
+    """
+    entitlements: Optional[Any] = None
+    """
+    (deprecated) Features to which the user is entitled within the application.
+    """
+
+
+class Org(JsonModel):
+    id: Optional[str] = Field(None, example="org-id")
+    """
+    The unique id
+    """
+    name: Optional[str] = Field(None, example="org-name")
+    """
+    The name of the organization
+    """
+    owner_id: Optional[str] = Field(None, alias="ownerId", example="user-id")
+    """
+    The userId of the organization owner
+    """
+
+
+class Workspace(JsonModel):
+    id: Optional[str] = Field(None, example="workspace-id")
+    """
+    The unique id
+    """
+    name: Optional[str] = Field(None, example="workspace-name")
+    """
+    The workspace name
+    """
+    enabled: Optional[bool] = Field(None, example=True)
+    """
+    Whether the workspace is enabled or not
+    """
+    default: Optional[bool] = Field(None, example=True)
+    """
+    Whether the workspace is the default. The default workspace is used when callers omit a workspace id
+    """
+
+
+class Error(JsonModel):
```

Comment:
We've already created the error model in the core package: https://github.com/ni/nisystemlink-clients-python/blob/master/nisystemlink/clients/core/_api_error.py
so this is unnecessary.

Reply:
These pydantic models follow the OpenApi yaml 1-1 and hence each model ends up with its copy of the API Error model as defined in the yaml.

Though logically correct to use the same Error model, it would not be practical to see what other models are common across clients and move them to the core for reuse.

Reply:
My thinking is that we should use the generated models as a starting point to save time, but not try to keep them perfectly in sync with the Swagger doc. We should be willing to make manual changes, such as consolidating duplicates and modifying the docstrings if needed to match specifics of the Python implementation. 

For example, here's an instance where we had to change `null` to `None`:

https://github.com/ni/nisystemlink-clients-python/blob/3459a508e43512b4e747d509c3ba58fd4c8f9aa5/nisystemlink/clients/dataframe/models/_modify_table_request.py#L33